### ISSUE TITLE
Create a proper development environment

### ISSUE THREAD

**Issue by audiodude** [openzim/wp1#137]
A few weeks ago I met with someone who was interested in collaborating on the project. However, he was immediately put off by the lack of a proper development environment. To tell the truth, I mostly depend on unit tests to make sure the code is correct (which is fine in and of itself), and when developing the web frontend, I use ssh tunnels to the live "production" databases (which is more problematic). Obviously this approach doesn't scale.

I picture the project having a docker-compose-dev.yml file that would spin up all the servers needed to run the project. Potentially, if users have their own access to enwiki_p, we could allow them to query that database and construct their own local copies of the wp10 db, even using Redis and the worker scripts.

Alternately, we could simply provide a docker-compose-dev.yml that spins up a test database that is already populated with some projects and data.

These are not mutually exclusive, in building the first option, it would produce data that could be used for the second.

**Comment by audiodude** [comment]
In the linked PR we implemented docker-compose-dev.yml which spins up a dev database that the development frontend can connect to.

I think the next step would be to somehow have a docker image of a working Mediawiki installation that was set up for API editing. That way, in development you could actually create and edit project tables.

**Comment by kelson42** [comment]
> I think the next step would be to somehow have a docker image of a working Mediawiki installation that was set up for API editing.

Something like this? https://hub.docker.com/r/openzim/mediawiki

**Comment by audiodude** [comment]
Yes that seems extremely useful! I would still have to make a new Docker image based on that image that had API editing enabled, and I'm not entirely sure how to do that. Still, the fact that the standard Mediawiki docker image requires interactive installation was a major hangup.

Thanks!

**Comment by kelson42** [comment]
@audiodude You can activate the API via the configuration file, something you need to have anyway. I don't think you need to create a new Docker image.

**Comment by audiodude** [comment]
Ah okay, thanks for the guidance!

**Comment by kelson42** [comment]
@audiodude I believe this issue woukd benefit of a small refresh

**Comment by audiodude** [comment]
What do you mean exactly? I think having a dev Mediawiki would be nice, but is less important since the bot is very mature now and it's the selection tools that we are mostly working on.

**Comment by kelson42** [comment]
@audiodude We could activate the API editing in our Mediawiki image... if this is what allows to close this issue. Is that necessary? Shoukd we open an issue there or just close this one?

**Comment by audiodude** [comment]
Sorry, I meant what do you mean by "small refresh".

Either way, I think this is s worthy idea, but there's no real pressing need to do it any time soon, so we can close if you like.

**Comment by audiodude** [comment]
Obsolete.
